hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractUpdateHandler.java

```diff
@@ -57,6 +77,19 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         if (op.hasAutoRenewAccountId() && !op.autoRenewAccountIdOrThrow().equals(AccountID.DEFAULT)) {
             context.requireKeyOrThrow(op.autoRenewAccountIdOrThrow(), INVALID_AUTORENEW_ACCOUNT);
         }
+
+        // TODO: Mono is using UpdateCustomizerFactory.processAdminKey but we don't have JContractIDKey/JKey
+        //  implementation in mod. So I stitched this together.
+        //  It's probably incorrect so I will wait for the reviews for guidance
+        if (op.hasAdminKey()) {
+            // TODO: is this correct? Is the contractID field deprecated?
+            if (op.adminKey().contractID() != null) {
+                throw new PreCheckException(INVALID_ADMIN_KEY);
+            }
+            if (op.adminKey().hasThresholdKey() && !op.adminKey().thresholdKey().hasKeys()) {
+                throw new PreCheckException(INVALID_ADMIN_KEY);
+            }
```

Comment:
Not sure if these checks are done

Reply:
Yes, I made those up based on the `givenAdminKeyMustBeValid` test. Now when I debug it in mono again I see it's failing in `processAdminKey`>`keyIfAcceptable`>`key.get() instanceof JContractIDKey`. Same reason the  `canMakeContractImmutableWithEmptyKeyList` and the `fridayThe13thSpec` tests are failing. The problem is that I don't know what is the alternative for `JContractIDKey` in **mod**(@Neeharika-Sompalli if you have some pointers here it will be very useful :))

Since I will be switching to enhancing `HapiTestEngine` stories @MiroslavGatsanoga will take over the failing tests.

For now, I will delete those checks and when we know what `JContractIDKey` is in **mod** we should be able to fix those 3 tests fairly easily.

Reply:
Alright, probably I will switch back to fixing the suites and will try `Key.contractID()`. Thanks